classpath/avian/FormatString.java

```diff
@@ -0,0 +1,992 @@
+      switch (c) {
+      case '$':
+      case '<':
+      case '0':
+      case '1':
+      case '2':
+      case '3':
+      case '4':
+      case '5':
+      case '6':
+      case '7':
+      case '8':
+      case '9':
+      case '-':
+      case '+':
+      case '\'':
+      case ' ':
+      case '#':
+      case ',':
+      case '.':
+      case '(':
+        return null;
+      // -- End of Flags and Switches ----------------------------------------
+
+      // -- Conversion characters --------------------------------------------
+      // If this point is reached, then the current character must be a valid
+      // conversion character. If it is not, it should fall through the rest
+      // of the switch statement below and throw an IllegalFormatException
+
+      // string conversion
+      case 'S':
+        setFlagTrue(FLAG_FORCE_UPPER_CASE);
+      case 's':
+        return seal(CONV_STRNG);
+
+      // newline conversion
+      case 'n':
+        return seal(CONV_NLINE);
+ 
+      // percent conversion
+      case '%':
+        return seal(CONV_PRCNT);
+      
+      // decimal numeric conversion
+      case 'd':
+        return seal(CONV_DECML);
+
+      // hexidecimal numeric conversion
+      case 'X':
+        setFlagTrue(FLAG_FORCE_UPPER_CASE);
+      case 'x':
+        return seal(CONV_HXDEC);
+ 
+      // datetime conversion
+      case 'T':
+        setFlagTrue(FLAG_FORCE_UPPER_CASE);
+      case 't':
+        _conversion = CONV_DTIME;
+        return null;
+
+      // boolean conversion
+      case 'B':
+        setFlagTrue(FLAG_FORCE_UPPER_CASE);
+      case 'b':
+        return seal(CONV_BOOLN);
+
+      // hashcode conversion
+      case 'H':
+        setFlagTrue(FLAG_FORCE_UPPER_CASE);
+      case 'h':
+        return seal(CONV_HCODE);
+
+      // character conversion
+      case 'C':
+        setFlagTrue(FLAG_FORCE_UPPER_CASE);
+      case 'c':
+        return seal(CONV_CHRCT);
+  
+      // octal numeric conversion
+      case 'o':
+        return seal(CONV_OCTAL);
+
+      // computerized scientific conversion
+      case 'E':
+        setFlagTrue(FLAG_FORCE_UPPER_CASE);
+      case 'e':
+        return seal(CONV_CPSCI);
+ 
+      // floating point conversion
+      case 'f':
+        return seal(CONV_FLOAT);
+
+      // general scientific floating point conversion
+      case 'G':
+        setFlagTrue(FLAG_FORCE_UPPER_CASE);
+      case 'g':
+        return seal(CONV_GNSCI);
+
+      // hexidecimal exponential floating point conversion
+      case 'A':
+        setFlagTrue(FLAG_FORCE_UPPER_CASE);
+      case 'a':
+        return seal(CONV_HXEXP);
+      // -- End of Conversion characters --------------------------------------
+     
+      default:
+        throw new IllegalFormatException(
+          "Invalid character encountered while parsing specifier: " + c);
+      }
+    }
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder();
```

Comment:
Yeah, no big deal.  I went ahead and merged it as-is, since I didn't notice any serious problems.  Feel free to open another pull request if you'd like.